speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/confscreens/ThresholdConfig.cpp

```diff
@@ -0,0 +1,202 @@
+#include <tgfclient.h>
+#include <iostream>
+#include "ThresholdConfig.h"
+#include "../legacymenu.h"
+#include "ConfigEnums.h"
+#include "Mediator.h"
+
+#define SND_PARAM_FILE    "config/Threshhold.xml"
+#define SND_SCT_THRESHOLD "Threshold Settings"
+#define SND_ATT_ACCEL     "Accel"
+#define SND_ATT_BRAKE     "Brake"
+#define SND_ATT_STEER     "Steer"
+
+static void* s_scrHandle = nullptr;
+static void* s_prevHandle = nullptr;
+
+// Data to store
+float m_accelVal = 0.0f;
+float m_brakeVal = 0.9f;
+float m_steerVal = 0.04f;
+
+// GUI settings Id's
+int m_accelThresholdControl;
+int m_brakeThresholdControl;
+int m_steerThresholdControl;
+
+/// @brief Handle input in the accel threshold textbox
+static void SetAccelThreshold(void*)
+{
+    // Get accel threshold from text box, clamped between 0 and 1
+    char* val = GfuiEditboxGetString(s_scrHandle, m_accelThresholdControl);
+    char* endptr;
+    m_accelVal = strtof(val, &endptr);
+    if (*endptr != '\0')
+        std::cerr << "Could not convert " << val << " to long and leftover string is: " << endptr << std::endl;
+
+    if (m_accelVal > 1.0f)
+        m_accelVal = 1.0f;
+    else if (m_accelVal < 0.0f)
+        m_accelVal = 0.0f;
```

Comment:
This is massive code duplication over all the functions below, so extract this part into a function.